### ISSUE TITLE
Support Calling removeParticipant by std::atexit [8984]

### ISSUE THREAD

**Issue by geoffviola** [eProsima/Fast-DDS#931]
Since a Participant relies on `thread_local` variables from https://github.com/eProsima/Fast-RTPS/commit/85ea0875c10bd1cd9b4a07afe90c1958d592bfe3, `std::atexit` cannot be used to shutdown a participant. The order of operations are
1. in scope
2. `thread_local` variable destructor
3. `std::atexit` call
4. `static` destructor

If `std::atexit` calls `removeParticipant`, it will try to access an out of scope variable.

Attached is a patch to reproduce the problem. You can see the issue by applying the patch to test/communication/Publisher.cpp and running SimpleCommunicationPublisher.
[publisher_at_exit.txt](https://github.com/eProsima/Fast-RTPS/files/3981985/publisher_at_exit.txt)

**Comment by MiguelCompany** [comment]
@geoffviola I am precisely working on using a pool of send buffers that will remove the `thread_local` variable.

You can take a look to [this branch](/eProsima/Fast-RTPS/tree/feature/message-groups-allocation/1.9.x)

**Comment by MiguelCompany** [comment]
@geoffviola This should have been solved after #938 and #1032. Could you check on your side?

**Comment by geoffviola** [comment]
Yes. It is working. Thanks for addressing this issue.
